examples/Middleware/Program.cs

```diff
@@ -5,47 +5,49 @@
 // -----------------------------------------------------------------------
 
 using System;
+using System.Threading.Tasks;
 using Proto;
 
 class Program
 {
     static void Main(string[] args)
     {
         var actor = Actor.FromFunc(c =>
-            {
-                Console.WriteLine($"actor got {c.Message.GetType()}:{c.Message}");
-                return Actor.Done;
-            })
-            .WithReceiveMiddleware(
-                next => async c =>
-                {
-                    Console.WriteLine($"middleware 1 enter {c.Message.GetType()}:{c.Message}");
-                    await next(c);
-                    Console.WriteLine($"middleware 1 exit");
-                },
-                next => async c =>
-                {
-                    Console.WriteLine($"middleware 2 enter {c.Message.GetType()}:{c.Message}");
-                    await next(c);
-                    Console.WriteLine($"middleware 2 exit");
-                })
-            .WithSenderMiddleware(
-                next => async (c, target, envelope) =>
-                {
-                    Console.WriteLine($"middleware 1 enter {c.Message.GetType()}:{c.Message}");
-                    await next(c, target, envelope);
-                    Console.WriteLine($"middleware 1 enter {c.Message.GetType()}:{c.Message}");
-                },
-                next => async (c, target, envelope) =>
-                {
-                    Console.WriteLine($"middleware 2 enter {c.Message.GetType()}:{c.Message}");
-                    await next(c, target, envelope);
-                    Console.WriteLine($"middleware 2 enter {c.Message.GetType()}:{c.Message}");
-                }
-            );
+                         {
+                             Console.WriteLine($"actor got {c.Message.GetType()}:{c.Message}");
+                             return Actor.Done;
+                         })
+                         .WithReceiveMiddleware(
+                             next => async c =>
+                             {
+                                 Console.WriteLine($"middleware 1 enter {c.Message.GetType()}:{c.Message}");
+                                 await next(c);
+                                 Console.WriteLine($"middleware 1 exit {c.Message.GetType()}:{c.Message}");
+                             },
+                             next => async c =>
+                             {
+                                 Console.WriteLine($"middleware 2 enter {c.Message.GetType()}:{c.Message}");
+                                 await next(c);
+                                 Console.WriteLine($"middleware 2 exit {c.Message.GetType()}:{c.Message}");
+                             });
 
         var pid = Actor.Spawn(actor);
-        pid.Tell("hello");
+
+        var root = new RootContext(new MessageHeader(), next => async (c, target, envelope) =>

```

Comment:
Outbound middleware is set on the root context, so when this example Tells the actor, the middleware is activated for the message.
Inbound middleware in the actor then reacts to the incoming message.